recipes/folly/all/conanfile.py

```diff
@@ -167,25 +174,23 @@ def package_info(self):
             "openssl::openssl",
             "snappy::snappy",
             "zlib::zlib",
-            "zstd::zstd"
+            "zstd::zstd",
+            "libdwarf::libdwarf",
+            "libsodium::libsodium",
+            "xz_utils::xz_utils"
         ]
-        if Version(self.version) >= "2019.01.01.00":
-            self.cpp_info.components["libfolly"].requires.extend([
-                "libdwarf::libdwarf",
-                "libsodium::libsodium",
-                "xz_utils::xz_utils"
-            ])
-            if self.settings.os == "Linux":
-                self.cpp_info.components["libfolly"].requires.extend([
-                    "libiberty::libiberty",
-                    "libunwind::libunwind"
-                ])
-        if Version(self.version) >= "2020.08.10.00":
-            self.cpp_info.components["libfolly"].requires.append("fmt::fmt")
         if self.settings.os == "Linux":
+            self.cpp_info.components["libfolly"].requires.extend(["libiberty::libiberty", "libunwind::libunwind"])
             self.cpp_info.components["libfolly"].system_libs.extend(["pthread", "dl", "rt"])
+
+        if Version(self.version) >= "2020.08.10.00":
+            self.cpp_info.components["libfolly"].requires.append("fmt::fmt")
+            if self.settings.os == "Linux":
+                self.cpp_info.components["libfolly"].defines.append("FOLLY_HAVE_ELF")
+
         elif self.settings.os == "Windows":
             self.cpp_info.components["libfolly"].system_libs.extend(["ws2_32", "Iphlpapi", "Crypt32"])
```

Comment:
To make mingw@Linux people happier.
```suggestion
            self.cpp_info.components["libfolly"].system_libs.extend(["ws2_32", "iphlpapi", "crypt32"])
```